crates/aptos-crypto/src/bls12381.rs

```diff
@@ -0,0 +1,381 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides API for the BLS12-381 signature library (https://github.com/supranational/blst).
+
+use crate::{
+    hash::CryptoHash, signing_message, CryptoMaterialError, Length, PrivateKey, PublicKey,
+    Signature, SigningKey, Uniform, ValidCryptoMaterial, ValidCryptoMaterialStringExt,
+    VerifyingKey,
+};
+use anyhow::{anyhow, Result};
+use aptos_crypto_derive::{DeserializeKey, SerializeKey, SilentDebug, SilentDisplay};
+use blst::{blst_scalar, BLST_ERROR};
+use rand_core::{OsRng, RngCore};
+use serde::Serialize;
+use std::convert::TryFrom;
+
+const DST: &[u8] = b"BLS_SIG_BLS12381G1_XMD:SHA-256_SSWU_RO_NUL_";
```

Comment:
So this is this DST is for the min-signature-size variant. See here: https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-bls-signature#section-4.2.1

However, we want the min-PK variant, so I will change this.